net-p2p/dogecoin-qt/dogecoin-qt-1.14.6.ebuild

```diff
@@ -0,0 +1,83 @@
+# Copyright 2023 Gentoo Authors 
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+inherit desktop
+DESCRIPTION="Dogecoin Core Qt 1.14.6 (with Graphical User Interface)"
+HOMEPAGE="https://github.com/dogecoin"
+SRC_URI="https://github.com/dogecoin/dogecoin/archive/refs/tags/v${PV}.tar.gz -> ${PN}-v${PV}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+DB_VER="5.3"
+KEYWORDS="~amd64 ~x86"
+IUSE="tests +wallet zmq"
+DOGEDIR="/opt/${PN}"
+DEPEND="
+	dev-libs/libevent:=
+	dev-libs/protobuf
+	dev-libs/openssl
+	sys-devel/libtool
+	sys-devel/automake:=
+	>=dev-libs/boost-1.81.0-r1
+	wallet? ( sys-libs/db:"${DB_VER}"=[cxx] )
+	dev-qt/qtcore
+	dev-qt/qtgui
+	dev-qt/qtwidgets
+	dev-qt/qtdbus
+	dev-qt/qtnetwork
+	dev-qt/qtprintsupport
+	dev-qt/linguist-tools:=
+	wallet? ( media-gfx/qrencode )
+	zmq? ( net-libs/cppzmq )
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	sys-devel/autoconf
+	sys-devel/automake
+"
+
+	PATCHES=(
+		"${FILESDIR}"/"${PV}"-net_processing.patch
+		"${FILESDIR}"/"${PV}"-paymentserver.patch
+		"${FILESDIR}"/"${PV}"-transactiondesc.patch
+		"${FILESDIR}"/"${PV}"-deque.patch
+		"${FILESDIR}"/"${PV}"-gcc13.patch
+	)
+
+WORKDIR_="${WORKDIR}/dogecoin-${PV}"
+S=${WORKDIR_}
+
+src_configure() {
+	chmod 755 ./autogen.sh
+	./autogen.sh || die "autogen failed"
+	local my_econf=(
+		--enable-cxx
+		$(use_with wallet incompatible-bdb)
+		--bindir="${DOGEDIR}/bin"
+		CPPFLAGS="-I/usr/include/db${DB_VER}"
+		CFLAGS="-I/usr/include/db${DB_VER}"
```

Comment:
ebuilds need to respect user's CFLAGS, and why is this duplicated in CPPFLAGS also?

Reply:
> why is this duplicated in CPPFLAGS also?

I used it as a "safety net" to ensure the right db header files are used and software builds correctly.